testsys/src/run_aws_ecs.rs

```diff
@@ -111,51 +144,199 @@ impl RunAwsEcs {
             .unwrap_or(&self.cluster_name);
         let ec2_resource_name = self
             .ec2_resource_name
+            .clone()
             .unwrap_or(format!("{}-instances", self.cluster_name));
         let aws_secret_map = self.aws_secret.as_ref().map(|secret_name| {
             btreemap! [ AWS_CREDENTIALS_SECRET_NAME.to_string() => secret_name.clone()]
         });
 
-        let ecs_resource = Resource {
+        let ecs_resource = self.ecs_resource(cluster_resource_name, aws_secret_map.clone())?;
+
+        let ec2_resource = self.ec2_resource(
+            &ec2_resource_name,
+            aws_secret_map.clone(),
+            cluster_resource_name,
+        )?;
+
+        let tests = if self.upgrade_downgrade {
+            if let (Some(starting_version), Some(upgrade_version), Some(migration_agent_image)) = (
+                self.starting_version.as_ref(),
+                self.upgrade_version.as_ref(),
+                self.migration_agent_image.as_ref(),
+            ) {
+                let tuf_repo = if let (Some(tuf_repo_metadata_url), Some(tuf_repo_targets_url)) = (
+                    self.tuf_repo_metadata_url.as_ref(),
+                    self.tuf_repo_targets_url.as_ref(),
+                ) {
+                    Some(TufRepoConfig {
+                        metadata_url: tuf_repo_metadata_url.to_string(),
+                        targets_url: tuf_repo_targets_url.to_string(),
+                    })
+                } else {
+                    None
+                };
+                let init_test_name = format!("{}-initial-test", self.name);
```

Comment:
Suggesting these names for brevity and so that they will sort in the order that they execute:

```
format!("{}-1-initial", self.name);
format!("{}-2-migrate", self.name);
format!("{}-3-migrated", self.name);
format!("{}-4-migrate", self.name);
format!("{}-5-final", self.name);
```